### PR TITLE
new update for function generate_doc

### PR DESCRIPTION
**Pull request by lamrani002**
# NEW|New update for generate doc when delete object and rewrite properties

### CODE REVIEW

File: htdocs/core/lib/modulebuilder.lib.php [hunk 1]
```diff
@@ -663,8 +663,7 @@ function writePropsInAsciiDoc($file, $objectname, $destfile)
 		$string = trim($string, "'");
 		$string = rtrim($string, ",");
 
-		$array = [];
-		eval("\$array = [$string];");
+		$array = eval("return [$string];");
```
Comment (eldy) [inline htdocs/core/lib/modulebuilder.lib.php:666]: Try to use dol_eval()

Comment (lamrani002) [reply htdocs/core/lib/modulebuilder.lib.php:666]: with do_eval it's not the result I want, but I've taken the necessary steps to work with eval by cleaning the string and checking if it's an array

Comment (eldy) [reply htdocs/core/lib/modulebuilder.lib.php:664]: Try 
dol_eval("return $string;", 1);
and better
dol_eval("$string");

Comment (lamrani002) [reply htdocs/core/lib/modulebuilder.lib.php]: do_eval not work for me so i found other way to do it

File: dev/initdemo/mysqldump_dolibarr_17.sql.zip
Comment (eldy) [inline dev/initdemo/mysqldump_dolibarr_17.sql.zip:1]: To remove

**Pull request closed without merging**